### PR TITLE
unit tests: Use unittest.skipUnless to simplify main()

### PR DESCRIPTION
**Pull request by Ericson2314**
This makes the testsuite work better with other test runners, like
pytest. This is important because better test runners are very useful to
development (e.g. avoiding running succeeding tests again and again),
even if we want to still support 0 dependency testing of Meson though
keeping the default test runnner working.

CC @nirbheek @dcbaker

### CODE REVIEW

File: run_unittests.py [hunk 1]
```diff
@@ -5655,26 +5666,12 @@ def unset_envs():
...
-            cases += ['LinuxCrossArmTests']
-        if should_run_cross_mingw_tests():
-            cases += ['LinuxCrossMingwTests']
-    if is_windows() or is_cygwin():
-        cases += ['WindowsTests']
-    if is_osx():
-        cases += ['DarwinTests']
+             'PythonTests', 'NativeFileTests', 'RewriterTests', 'CrossFileTests',
+             'LinuxlikeTests', 'LinuxCrossArmTests', 'LinuxCrossMingwTests',
+             'WindowsTests', 'DarwinTests' ]
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[Flake8]__ [E202] whitespace before ']'

<sub>posted by Sider</sub>

**Comment by Ericson2314** [comment]
I think those Windows failures are spurious?

**Comment by nirbheek** [comment]
Working on a fix for the Cygwin failure in https://github.com/mesonbuild/meson/pull/4967. The VS 2015 failure is weird and also spurious. However, we need to fix those separately before we can merge any MRs...

**Comment by Ericson2314** [comment]
@nirbheek thanks for investigating! Pitty this stuff can change out from under us.

**Comment by Ericson2314** [comment]
Yay test suite is fixed! Rebased and this passes now.

**Pull request merged**